src/modules/hostname.rs

```diff
@@ -40,11 +40,16 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     } else {
         host.as_ref()
     };
+    let (_choose_style, choose_config) = if ssh_connection.is_some() {
+        ("ssh_style", config.ssh_style)
+    } else {
+        ("style", config.style)
+    };
 
     let parsed = StringFormatter::new(config.format).and_then(|formatter| {
         formatter
             .map_style(|variable| match variable {
-                "style" => Some(Ok(config.style)),
+                _choose_style => Some(Ok(choose_config)),
```

Comment:
This likely doesn't work the way you think it does. This is always going to match, shadowing whatever value `_choose_style` had previously. The `_ => None` arm after this will never be matched.

Here is one way to rewrite it, though I don't think it's optimal:

```
match variable {
  "ssh_style" if ssh_connection.is_some() => Some(Ok(config.ssh_style)),
  "ssh_style" if ssh_connection.is_none() => Some(Ok(config.style)),
  "style" => Some(Ok(config.style)),
  _ => None,
}
```